swap/types.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package swap
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// ChequeParams encapsulate all cheque parameters
+type ChequeParams struct {
```

Comment:
there's just too many files in this package. i find it very difficult to follow what should be where. can we please start with big monolithic files and break stuff out to other files when the need actually arises?

Reply:
i think we are leaning towards leaving the package as it is, since it would probably be too much of a change at this point in the process (but please correct me if i am wrong in assuming this).

however, i would like to ask: is this big monolithic file approach good golang style?

Reply:
@mortelli the division between files in this package is arbitrary. Types are practically everywhere (instead of being in `types.go`). I also don't understand why we need an entirely different file for `prices` - why do you need to add a whole file just to have 3 variable initializers in it? `protocol.go` does not handle any of the protocol messages (i would expect it to handle the protocol message exchange in its entirety, instead it just shows a partial image of what the protocol does).

> however, i would like to ask: is this big monolithic file approach good golang style?

i've seen quite a lot of other golang projects using this approach. 
to further elaborate, i'd say that separating into different files is fine, as long as the separation serves a purpose, but when you force a large chunk of code into different files just in order to split it into multiple files - it doesn't. please reflect on the content of the files in this directory and honestly say if the division into `types, protocol, cheque, config, oracle, peer, prices, swap` makes sense. i see maybe justification to have `cheque, peer, swap` and maybe `config` if you really insist.

Reply:
@acud i see what you mean and i think it's a valid concern. i'd like to discuss this with the rest of the incentives team and get back to you on this point.

Reply:
@acud `prices` is there because it is currently a thing which is very temporary. Prices should and will not be kept hardcoded in a file in the future. We are in the process of creating SWIPs which will introduce just that, the official mechanism through which we will set prices, which will be through on-chain oracles, and that will make having hardcoded prices obsolete. At that point, the complete `prices` file will be removed. 

In this view I think it is absolutely justified to have them in a separate file. And not only that. Prices are something which conceptually have nothing to do with the code, they are a "business" entity so to speak. I think this very much justifies them to be clearly separated.

`oracle` is the same story. It is a placeholder for something we just don't have yet and will be implemented with the next SWIP. We can't wait for that or we have an even bigger PR. We can also not do without because otherwise we have to have convoluted code for something which will be replaced and this way at least we have some kind of clean interface. It definitely should stay in its own file.

Why do you have a problem with `protocol.go`, if that is a common pattern in our code? `swap.go` is 586 lines long, `protocol.go` another 165, why should I want to have a file with thousands of lines? `protocol.go` contains the definitions of `Protocols` and `APIs` etc. I think it is a nice conceptual separation, it is not arbitrary. I don't see why you want it to be removed. There also is a historical reason for this, in that there was initially a separate object as the protocol, but this got refactored in Balaton, putting moving now all the methods therein to `Swap`. This may possibly now make it look like arbitrary to you because it's the same receiver now. 

Finally there is `types.go`, which indeed is a bit of a small file, but this is a pattern you can find throughout the complete `go-ethereum` codebase, and also swarm: `pss`, `storage`,`simulation`.

I think this is a thing of personal preference and I don't agree to your requirement in this case. I personally do like to split things and not have thousands of lines in a single file. I would, in all respect, want to insist to keep most as-is, or maybe we can move everything from types into swap.

Reply:
@holisticode I find it not so efficient to be navigating between files just to find one variable/type declaration. The separation of `protocol.go` is not needed in my opinion and I did not by any way conclude that I agree with having this filenaming pattern across our codebase, I dislike it and I will try as much as I can to not use this pattern because I find it confusing and counter-intuitive (I don't want to start thinking which file to open in order to access which method on the same receiver type.... you catch my drift).
You can also evidently look at the current stream package to see how much of a mess of splitting stuff up into files has caused. It is virtually impossible to find anything in that package. To an extent it seems that the filenames are merely coincidental.
If you are adamant on having things this way - have it your way. I'm resolving this